### DISCUSSION TITLE
替换图片后，帧渲染时耗急剧增加，怎么解决？

### DISCUSSION THREAD

**Discussion by SilenSang** [Tencent/libpag#2235] [Q&A] [answered]
我的程序需要在渲染的过程中，随着时间去替换一些图片。没有发生替换的帧，渲染时间约 0.04 毫秒，极快；但发生替换的帧，渲染时间骤增至平均 100 多毫秒，最长达到了 200 多毫秒。请问这个有办法优化加快吗？

**Comment by kevingpqi123** [comment, marked as answer]
其实这里的耗时主要是图片解码，渲染倒不是太耗时，图片解码的操作你们可以先异步处理好，渲染当前帧的时候，获取到图片的 RGBA 数据，然后构造 PAGImage 完成占位图替换操作
